spock-specs/src/test/groovy/org/spockframework/smoke/condition/SetComparisonRendering.groovy

```diff
@@ -0,0 +1,111 @@
+package org.spockframework.smoke.condition
+
+class SetComparisonRendering extends ConditionRenderingSpec {
+
+  def "Set differences are displayed"() {
+    expect:
+    isRendered('''
+a == b
+| |  |
+| |  [2, 3, 4]
+| false
+| 2 differences (66% similarity missing 1 extra 1)
+| missing: [4]
+| extra: [1]
+[1, 2, 3]
+''') {
+      Set a = [1, 2, 3]
+      Set b = [2, 3, 4]
+      assert a == b
+    }
+  }
+
+  def "Set difference is displayed"() {
+    expect:
+    isRendered('''
+a == b
+| |  |
+| |  [1, 2, 3, 4]
+| false
+| 1 difference (75% similarity missing 1 extra 0)
+| missing: [4]
+| extra: []
+[1, 2, 3]
+''') {
+      Set a = [1, 2, 3]
+      Set b = [1, 2, 3, 4]
+      assert a == b
+    }
+  }
+
+  def "Set differences for larger Set with small differences"() {
+    expect:
+    isRendered('''
+a == b
+| |  |
+| |  [6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
+| false
+| 10 differences (66% similarity missing 5 extra 5)
+| missing: [16, 17, 18, 19, 20]
+| extra: [1, 2, 3, 4, 5]
+[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
+''') {
+      Set a = (1..15) as Set
+      Set b = (6..20) as Set
+      assert a == b
+    }
+  }
+
+  def "Set differences for larger Set with large differences"() {
+    expect:
+    isRendered('''
+a == b
+| |  |
+| |  [16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30]
+| false
+| 30 differences (0% similarity missing 15 extra 15 details omitted)
```

Comment:
I would put the `details ommited` in the line where otherwise `missing:` was, because there you look for it.
Alternative:
```
30 differences (0% similarity missing 15 extra 15)
missing/extra: <too many to render|details ommited>
```